parquet/spec.go

```diff
@@ -1,21 +1,98 @@
 package parquet
 
-import "github.com/invopop/jsonschema"
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/apache/arrow/go/v17/parquet"
+	"github.com/invopop/jsonschema"
+)
+
+var allowedVersions = []string{"v1.0", "v2.4", "v2.6", "v2Latest"}
+var allowedRootRepetitions = []string{"undefined", "required", "optional", "repeated"}
 
 // nolint:revive
-type ParquetSpec struct{}
+type ParquetSpec struct {
+	Version        string `json:"version,omitempty"`
+	RootRepetition string `json:"root_repetition,omitempty"`
+}
+
+func (s *ParquetSpec) GetVersion() parquet.Version {
+	switch s.Version {
+	case "v1.0":
+		return parquet.V1_0
+	case "v2.4":
+		return parquet.V2_4
+	case "v2.6":
+		return parquet.V2_6
+	case "v2Latest":
```

Comment:
Should this be a `default`?

Reply:
That's the current default

Reply:
I was referring to syntax rather than semantics. The last two return statements are identical.

Reply:
If you think it's clearer, that's fair enough.

Reply:
> I was referring to syntax rather than semantics. The last two return statements are identical.

Ah got it, yeah I was unsure which way to go. I don't have a strong opinion so I'll update the PR

Reply:
Ah OK so the reason I did it like this is otherwise we have:
```go
	switch s.Version {
	case "v1.0":
		return parquet.V1_0
	case "v2.4":
		return parquet.V2_4
	case "v2.6":
		return parquet.V2_6
	default:
		return parquet.V2_LATEST
	}
```

and if we ever change the `default` part it's easy to miss add back `case v2Latest`

Reply:
Fair enough